### PR TITLE
[release/0.20] alpine: avoid wiping out writable host mounts under /home, etc

### LINKED ISSUE #2221: Lost files on host when canceling limactl start
### Description

Canceling `limactl start debug` deleted files that were mounted from the host.

I don't have the exact config anymore but it was similar to this. The mounted ~/Files directory was deleted.
```yaml
images:
  - location: "https://github.com/lima-vm/alpine-lima/releases/download/v0.2.35/alpine-lima-std-3.19.0-aarch64.iso"
    arch: "aarch64"
    digest: "sha512:13e50601ee65af5d7a6dfd30bb41fd89f8bf806ecdb516c61fe238c3cf3b57cf67469418a99f329bb4c343e3387e6e0fd4fe20501cfd501f031f7244adc67215"

containerd:
  system: false
  user: false
  archives:
    - location: "https://github.com/containerd/nerdctl/releases/download/v2.0.0-beta.0/nerdctl-full-2.0.0-beta.0-linux-arm64.tar.gz"
      arch: "aarch64"
      digest: "sha256:29c58dc39ebca1642f14827915f67d909ec49b1d7458938401cf835edbc5b8d0"

os: "Linux"
arch: "aarch64"
vmType: "vz"
cpus: 2
memory: "2GiB"
disk: "2GiB"

mountType: "virtiofs"
mounts:    
  - location: "~/Files"                                                                                                                      
    mountPoint: "/Files"    
    writable: true  

audio:
  device: "vz"
video:
  display: "none"
networks:
  - vzNAT: true
ssh:
  localPort: 54201
  loadDotSSHPubKeys: false
  forwardAgent: false
  forwardX11: false
  forwardX11Trusted: false
```

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Cherry-pick (clean):
- #2238

> * Fix parsing ssh keys as block string
> * Create a mount script instead of editing /etc/fstab
> * Make lima-init.sh yaml parsing more robust
> 

- #2234

> A host directory could be wiped out when all the following conditions are met:
> - The directory is mounted to Lima via virtiofs or 9p (reverse-sshfs is not affected)
> - The mount is writable
> - The mount point in the guest is under one of: /etc /home /root /usr/local /var/lib
> - The guest OS is Alpine Linux
> 
> Fix #2221
> Fix rancher-sandbox/rancher-desktop#6582

- #2242
> The escaping is needed to print the string with quotes, but would break the umount command.

### CODE REVIEW

File: pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh [hunk 1]
```diff
@@ -9,6 +9,36 @@ test -f /etc/alpine-release || exit 0
...
+		MNTTYPE="$(echo "${LINE}" | awk '{print $3}')"
+		[ "${MNTTYPE}" = "ext4" ] && continue
+		[ "${MNTTYPE}" = "tmpfs" ] && continue
+		MNTOPTS="$(echo "${LINE}" | awk '{print $4}')"
+		MNTPNT=${MNTPNT//\\/\\\\}
+		MNTPNT=${MNTPNT//\"/\\\"}
+		echo "mount -t \"${MNTTYPE}\" -o \"${MNTOPTS}\" \"${MNTDEV}\" \"${MNTPNT}\"" >>/mnt.sh
+		# Before mv, unmount filesystems (virtiofs, 9p, etc.) below "${DIR}", otherwise host mounts will be wiped out
+		# https://github.com/rancher-sandbox/rancher-desktop/issues/6582
+		umount "${MNTPNT}" || exit 1
```
Comment (jandubois) [inline pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh:37]: Sorry, I missed this in my suggestion to fix backslashes and double-quotes for the `echo` statement: the `umount` needs to be executed before those substitutions.

Comment (jandubois) [reply pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh:37]: See #2242

Comment (AkihiroSuda) [reply pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh:34]: cherry-picked

**Comment by AkihiroSuda** [comment]
Ah, accidentally merged this to `master` 🤦

**Comment by AkihiroSuda** [comment]
Reopened as :
- https://github.com/lima-vm/lima/pull/2244

**Pull request merged**